### PR TITLE
[WIP] Add minimal iso

### PR DESCRIPTION
**Pull request by carbonin**
The first part of this PR extracts the iso handling packages from `assisted-service` with all their associated git history and adds them to this project.
I did this to ensure that we don't lose valuable reasoning behind any of the choices we previously made while developing this feature.

The second part of the PR adds commits to remove some things we won't be using in this project as well as re-work the existing code to be more focused.
Specifically, the iso editor factory is no longer needed as that was created primarily for throttling multiple minimal iso customizations. In this project that will all be done using the "streaming" solution so we don't need any of the customization details from these packages.

The rest of the PR adds a query parameter to determine whether the minimal or full iso is being requested.

### CODE REVIEW

File: internal/isoeditor/rhcos.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	"github.com/carbonin/assisted-image-service/internal/isoutil"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	RamDiskPaddingLength  = uint64(1024 * 1024) // 1MB
+	IgnitionPaddingLength = uint64(256 * 1024)  // 256KB
+	ignitionImagePath     = "/images/ignition.img"
+	ramDiskImagePath      = "/images/assisted_installer_custom.img"
```
Comment (djzager) [inline internal/isoeditor/rhcos.go:21]: This is what we will be asking the assisted-service for right? Wonder if it should just be 
```suggestion
	ramDiskImagePath      = "/images/ramdisk.img"
```

**Review by djzager — COMMENTED** [review]
Just one comment. Did my best to visually inspect it. Nothing stood out as crazy.

> The rest of the PR adds a query parameter to determine whether the minimal or full iso is being requested.

I did not see this change, would have expected it in the handler somewhere. Did I miss it?

**Comment by carbonin** [comment]
> I did not see this change, would have expected it in the handler somewhere. Did I miss it?

Nope, you're right. I haven't added that bit yet, that's mainly what the WIP is for :laughing:

File: internal/isoeditor/rhcos.go (hunk 1 above)
Comment (carbonin) [reply internal/isoeditor/rhcos.go:21]: :man_shrugging: This isn't a new name and I'm not sure it really matters

**Comment by carbonin** [comment]
Repo has moved to https://github.com/openshift/assisted-image-service

**Pull request closed without merging**